docs/develop/python/interceptors.mdx

```diff
@@ -0,0 +1,120 @@
+---
+id: interceptors
+title: Interceptors - Python SDK
+sidebar_label: Interceptors
+description: Learn how to implement Interceptors in the Temporal Python SDK to manage inbound and outbound SDK calls, enhance tracing, and add authorization to your Workflows and Activities.
+toc_max_heading_level: 2
+keywords:
+  - interceptors
+tags:
+  - Interceptors
+  - Python SDK
+  - Temporal SDKs
+---
+
+### Interceptors
+
+The behavior of the Python SDK can be customized in many useful ways by modifying inbound and outbound calls using
+interceptors. This is similar to the use of middleware in other frameworks.
+
+There are five categories of inbound and outbound calls that you can modify in this way:
+
+1. Outbound Client calls, such as `start_workflow()`, `signal_workflow()`, `list_workflows()`, `update_schedule()`, etc.
+
+2. Inbound Workflow calls: `execute_workflow()`, `handle_signal()`, `handle_update_handler()`, etc
+
+3. Outbound Workflow calls: `start_activity()`, `start_child_workflow()`, `start_nexus_operation()`, etc
+
+4. Inbound call to execute an Activity: `execute_activity()`
+
+5. Outbound Activity calls: `info()` and `heartbeat()`
+
+To modify outbound Client calls, define a class inheriting from
+[`client.Interceptor`](https://python.temporal.io/temporalio.client.Interceptor.html), and implement the method
+`intercept_client()` to return an instance of
+[`OutboundInterceptor`](https://python.temporal.io/temporalio.client.OutboundInterceptor.html) that implements the
+subset of outbound Client calls that you wish to modify.
+
+Then, pass a list containing an instance of your `client.Interceptor` class as the
+`interceptors` argument of [`Client.connect()`](https://python.temporal.io/temporalio.client.Client.html#connect).
+
+The purpose of the interceptor framework is that the methods you implement on your interceptor classes can perform
+arbitrary side effects and/or arbitrary modifications to the data, before it is received by the SDK's "real"
```

Comment:
Is "data" what we mean here? I think it'd be best to focus on the side effects case first, otherwise someone might get mixed up about these being primarily useful for data transformations a la data converters.

Reply:
Yes, please go ahead and rewrite.